nifi-registry/nifi-registry-core/nifi-registry-jetty/src/main/java/org/apache/nifi/registry/jetty/handler/StandardHandlerProvider.java

```diff
@@ -0,0 +1,257 @@
+
+    private static final String OIDC_SUPPORTED_PARAMETER = "oidc-supported";
+
+    private static final String PROPERTIES_PARAMETER = "nifi-registry.properties";
+
+    private static final String KEY_PROVIDER_PARAMETER = "nifi-registry.key";
+
+    private static final String RESOURCE_BASE_PARAMETER = "resourceBase";
+
+    private static final String DIR_ALLOWED_PARAMETER = "dirAllowed";
+
+    private static final String WEB_INF_JAR_PATTERN_ATTRIBUTE = "org.eclipse.jetty.server.webapp.WebInfIncludeJarPattern";
+
+    private static final String WEB_INF_JAR_PATTERN = ".*/spring-[^/]*\\.jar$";
+
+    private final CryptoKeyProvider cryptoKeyProvider;
+
+    private final String docsDirectory;
+
+    public StandardHandlerProvider(final CryptoKeyProvider cryptoKeyProvider, final String docsDirectory) {
+        this.cryptoKeyProvider = Objects.requireNonNull(cryptoKeyProvider, "Key Provider required");
+        this.docsDirectory = docsDirectory;
+    }
+
+    /**
+     * Get Jetty Handler for Registry Server containing mappings to web applications
+     *
+     * @param properties Registry properties
+     * @return Jetty Handler
+     */
+    @Override
+    public Handler getHandler(final NiFiRegistryProperties properties) {
+        Objects.requireNonNull(properties, "Properties required");
+
+        final File libDirectory = properties.getWarLibDirectory();
+        final File workDirectory = properties.getWebWorkingDirectory();
+
+        final HandlerCollection handlers = new HandlerCollection();
+        // Add Header Writer Handler before others
+        handlers.addHandler(new HeaderWriterHandler());
+
+        final WebAppContext userInterfaceContext = getWebAppContext(libDirectory, workDirectory, ClassLoader.getSystemClassLoader(), UI_FILE_PATTERN, UI_CONTEXT_PATH);
+        userInterfaceContext.setInitParameter(OIDC_SUPPORTED_PARAMETER, Boolean.toString(properties.isOidcEnabled()));
+        handlers.addHandler(userInterfaceContext);
+
+        final ClassLoader apiClassLoader = getApiClassLoader(properties.getDatabaseDriverDirectory());
+        final WebAppContext apiContext = getWebAppContext(libDirectory, workDirectory, apiClassLoader, API_FILE_PATTERN, API_CONTEXT_PATH);
+        apiContext.setAttribute(PROPERTIES_PARAMETER, properties);
+        apiContext.setAttribute(KEY_PROVIDER_PARAMETER, cryptoKeyProvider);
+        handlers.addHandler(apiContext);
+
+        final WebAppContext docsContext = getWebAppContext(libDirectory, workDirectory, ClassLoader.getSystemClassLoader(), DOCS_FILE_PATTERN, DOCS_CONTEXT_PATH);
+        final File docsDir = getDocsDir();
+        final ServletHolder docsServletHolder = getDocsServletHolder(docsDir);
+        docsContext.addServlet(docsServletHolder, HTML_DOCS_PATH);
+
+        final File apiDocsDir = getApiDocsDir(apiContext);
+        final ServletHolder apiDocsServletHolder = getDocsServletHolder(apiDocsDir);
+        docsContext.addServlet(apiDocsServletHolder, REST_API_DOCS_PATH);
```

Comment:
Something appears to be wrong with the docs:

<img width="1156" alt="image" src="https://github.com/apache/nifi/assets/423565/f894ac1f-0d97-4b1b-89f5-30d84736f7c0">

Reply:
Thanks for catching this problem @gresockj! The problem was due to using the same name for the regular documentation and REST API documentation Servlet Holders. As a result, the REST API Holder overwrote the regular documentation. I corrected the naming to use the directory path, which is unique, now all documentation pages render as expected.